crates/provider/src/utils.rs

```diff
@@ -39,6 +39,17 @@ pub fn eip1559_default_estimator(base_fee_per_gas: U256, rewards: &[Vec<U256>])
     (potential_max_fee, max_priority_fee_per_gas)
 }
 
+/// Identifies the intended transport type from the given string.
+pub fn parse_str_to_tranport_type(s: &str) -> &str {
+    if s.starts_with("http://") || s.starts_with("https://") {
+        "http"
```

Comment:
I think we also want to check for things like "localhost:port"
or if the input is a `SocketAddress`, then prefix with http,

so the Http variant should contain a String

Reply:
also need to extract any auth info for WsConnect

some useful methods:
https://docs.rs/url/latest/url/struct.Url.html#method.has_authority
https://docs.rs/url/latest/url/struct.Url.html#method.origin
https://docs.rs/url/latest/url/struct.Url.html#method.scheme

Reply:
do we have `parse` implementations on the relevant `Connect` structs?